orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -557,8 +557,7 @@ def _add_input(name, input, onnx_graph, onnx_graph_input_names):
         elif input_parameter.kind == inspect.Parameter.VAR_KEYWORD:
             # **kwargs is always the last argument of forward()
             for name, inp in kwargs.items():
-                if name not in input_names:
```

Comment:
OK, thanks for the fix. I just did some homework around how the python parameters work, 
Let me add some understanding for this issue for other reviewers to get the idea quickly. 

>>> def foo(*args, named_kwargs, **kwargs):
...           print("foo")
>>> from inspect import signature
>>> sig = signature(foo)
>>> ps = [p for p in sig.parameters.values()]
>>> ps[0].name, ps[0].kind
('args', <_ParameterKind.VAR_POSITIONAL: 2>)
>>> ps[1].name, ps[1].kind
('named_kwargs', <_ParameterKind.KEYWORD_ONLY: 3>)
>>> ps[2].name, ps[2].kind
('kwargs', <_ParameterKind.VAR_KEYWORD: 4>)

So `named_kwargs` is a keyword_only argument from parameter inspection. 

While the caller of parse_inputs_for_onnx_export, e.g. (orttraining/orttraining/python/training/ortmodule/_training_manager.py.forward(self, *inputs, **kwargs)>>_export_model(self, *inputs, **kwargs)>>_get_exported_model(self, input_schema, *inputs, **kwargs) >> parse_inputs_for_onnx_export(all_input_parameters, onnx_graph, schema, inputs, kwargs)) looks parsed in inputs only containing VAR_POSITIONAL (*args) and VAR_KEYWORD (**kwargs, so named_kwargs is stored in this value).

So in the loop starting from line 534 of _io.py, it goes to "or input_parameter.kind == inspect.Parameter.KEYWORD_ONLY" to handle the "named_kwargs" once, then later, goes to "elif input_parameter.kind == inspect.Parameter.VAR_KEYWORD:" to handle the "named_kwargs" the second time.

Reply:
The change looks reasonable to me, @baijumeswani @thiagocrepaldi, could you please share your options on this? Thanks!

Reply:
Looks good. Thanks.